test/components/syncTest.js

```diff
@@ -0,0 +1,788 @@
+      })
+      .waitUntil(function () {
+        return this.getText('.bookmarkToolbarButton:nth-child(3) .bookmarkText')
+          .then((title) => title === folder1Title)
+      })
+  })
+})
+
+describe('Syncing history', function () {
+  Brave.beforeAllServerSetup(this)
+
+  function * setup (seed) {
+    yield Brave.startApp()
+    yield setupBrave(Brave.app.client)
+    // New tab sites appear in history; so clear them out.
+    yield Brave.app.client
+      .onClearBrowsingData({browserHistory: true})
+      .changeSetting(settings.NEWTAB_MODE, newTabMode.EMPTY_NEW_TAB)
+      .changeSetting(settings.SYNC_TYPE_HISTORY, true)
+    yield setupSync(Brave.app.client, seed)
+  }
+
+  before(function * () {
+    this.page1Url = Brave.server.url('page1.html')
+    this.page1Title = 'Page 1'
+    this.page2Url = Brave.server.url('page2.html')
+    this.page2Title = 'Page 2'
+    this.page3Url = Brave.server.url('fingerprinting.html')
+    this.page3Title = this.page3Url
+    this.seed = new Immutable.List(crypto.randomBytes(32))
+    yield setup(this.seed)
+
+    // For Create: Visit page 1
+    yield Brave.app.client
+      .tabByIndex(0)
+      .loadUrl(this.page1Url)
+
+    // For order: Visit page 2
+    yield Brave.app.client
+      .tabByIndex(0)
+      .waitForUrl(this.page1Url)
+      .loadUrl(this.page2Url)
+
+    // Private tab should not sync
+    yield Brave.app.client
+      .waitForUrl(this.page2Url)
+      .windowParentByUrl(this.page2Url)
+      .ipcSend(messages.SHORTCUT_NEW_FRAME, this.page3Url, { isPrivate: true })
+      .waitForUrl(this.page3Url)
+
+    // XXX: Wait for sync to upload records to S3
+    yield Brave.app.client.pause(1000)
+
+    // Finally start a fresh profile and setup sync
+    yield Brave.stopApp()
+    yield setup(this.seed)
+    yield Brave.app.client
+      .tabByIndex(0)
+      .url(aboutHistoryUrl)
+  })
+
+  after(function * () {
+    yield Brave.stopApp()
+  })
+
+  it('create', function * () {
+    yield Brave.app.client
+      .waitForVisible(`table.sortableTable td.title[data-sort="${this.page1Title}"]`)
+      .waitForVisible(`table.sortableTable td.title[data-sort="${this.page2Title}"]`)
+  })
+
+  it('sync order', function * () {
+    const page1Title = this.page1Title
+    const page2Title = this.page2Title
+
+    yield Brave.app.client
+      .waitUntil(function () {
+        return this.getText('table.sortableTable tbody tr:nth-child(1) td.title')
+          .then((title) => title === page2Title)
+      })
+      .waitUntil(function () {
+        return this.getText('table.sortableTable tbody tr:nth-child(2) td.title')
+          .then((title) => title === page1Title)
+      })
+  })
+
+  it('private browsing does not sync', function * () {
+    yield Brave.app.client
+      .waitForElementCount('table.sortableTable tbody tr', 2)
+      .waitForVisible(`table.sortableTable td.title[data-sort="${this.page3Title}"]`, 1000, true)
+  })
+})
+
+describe('Syncing site settings', function () {
```

Comment:
i believe the record which is received later always wins. https://github.com/brave/sync/blob/staging/client/recordUtil.js#L103